tests/contracts/zil-game/contract.scilla

```diff
@@ -52,16 +52,16 @@ let time_to_claim =
     end     
 
 let check_validity = 
-  fun (a        : Address) =>
+  fun (a        : ByStr20) =>
   fun (solution : Int128) =>
-  fun (pa       : Address) =>
-  fun (pb       : Address) =>
-  fun (guess_a  : Option Hash) =>
-  fun (guess_b  : Option Hash) =>
+  fun (pa       : ByStr20) =>
+  fun (pb       : ByStr20) =>
+  fun (guess_a  : Option ByStr32) =>
+  fun (guess_b  : Option ByStr32) =>
```

Comment:
I

Reply:
The solution seems sound enough, but I wonder whether it is a good idea to remove the Address and Hash types from the scilla source language. I think it is a bad idea in general to force the user to know about how the language is implemented, so from the point of view of the user it is easier to work with Address/Hash as an abstract notion without having to deal with how addresses and hashes are represented in at runtime.

I believe we could achieve this by treating the Address and Hash types as syntactic sugar in the parser. Has this been considered?